File: 2022.1/Programação Orientada a Objetos/lista1python/Ex9.py
def f(x):
    a = 1
    b = -1
    c = -13
    d = 8

    return (a*x*x*x + b*x*x + c*x +d)

def f1(x):
    a = 3
    b = -2
    c = -13

    return (a*x*x + b*x + c)

x0 = float(input())

contador = 0

x1 = x0 - (f(x0)/f1(x0));

while(abs(x1-x0) > 0.0000001):

    x0 = x1
    x1 = x0 - (f(x0)/f1(x0));

    contador += 1

print("A raiz encontrada foi:", x1)
print("Foram realizadas", contador, "iterações")

File: 2022.1/Programação Orientada a Objetos/lista1java/Ex9.java
public class Ex9{
    public static double f1(double x){
        int a = 3;
        int b = -2;
        int c = -13;

        return (a*x*x + b*x + c);
    }

    public static double f(double x){
        int a = 1;
        int b = -1;
        int c = -13;
        int d = 8;

        return (a*x*x*x + b*x*x + c*x + d);
    }

    public static void main(String args[]) throws Exception{

        double x0 = EntradaTeclado.leDouble();
        double x1 = x0;
        
        int contador = 0;

        do{
            x0 = x1;
            x1 = x0 - (f(x0)/f1(x0));

            contador++;
        }while(Math.abs(x1 - x0) > 0.0000001);

        System.out.println("A raiz encontrada foi: " + x1);
        System.out.println("Foram realizadas " + contador + " iterações");
    }
}
